x-pack/plugins/apm/public/context/service_group/service_group_context.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useRef } from 'react';
+import { useLocation } from 'react-router-dom';
+import * as qs from 'query-string';
+
+export interface ServiceGroupContextValue {
+  reloadServiceGroupList: { current: () => void };
+  serviceGroupName: string;
+}
+
+export const ServiceGroupContext =
+  React.createContext<ServiceGroupContextValue>({
+    reloadServiceGroupList: { current: () => {} },
+    serviceGroupName: '',
+  });
+
+export function ServiceGroupProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const reloadServiceGroupList = useRef(() => {});
+  const { search } = useLocation();
+  const query = qs.parse(search);
+  const serviceGroupName = Array.isArray(query.serviceGroup)
+    ? query.serviceGroup[0]
+    : query.serviceGroup || '';
+
```

Comment:
I'm having a hard time understand why this needs to be a new context, and why we do custom url handling. Can't we use the [link client](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/routing_and_linking.md#client-side)?